portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -66,25 +72,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     public static class UserData {
         long id;
         String name;
-        String email;
         String comment;
         Date date;
+        String email;
 
-        UserData(long id, String name, String email, String comment, Date date) {
+        UserData(long id, String name, String comment, Date date, String email) {
             this.id = id;
             this.name = name;
-            this.email = email;
             this.comment = comment;
             this.date = date;
+            this.email = email;
         }
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Date date = new Date();
         String name = getParameter(request, "name", "");
-        String email = getParameter(request, "email", "");
         String comment = getParameter(request, "comment", "");
+        String userEmail = userService.getCurrentUser().getEmail();
```

Comment:
what happens if the current user isn't logged in?

Reply:
The comment form is not displayed when the user isn't logged in. I recall you mentioned I shouldn't assume how the front-end and back-end  interact. So, I added a conditional that checks if the user is logged in before posting to datastore.